ui/packages/consul-ui/app/components/topology-metrics/index.js

```diff
@@ -67,6 +67,27 @@ export default class TopologyMetrics extends Component {
   }
 
   // =actions
+  @action
+  setHeight() {
+    if (document.querySelector('#downstream-container')) {
+      const downContainer = document.querySelector('#downstream-container').getBoundingClientRect();
+
+      document
+        .querySelector('#downstream-lines')
+        .setAttribute('style', `height:${downContainer.height}px`);
+    }
+
+    if (document.querySelector('#upstream-container')) {
+      const upContainer = document.querySelector('#upstream-container').getBoundingClientRect();
+
+      document
+        .querySelector('#upstream-lines')
+        .setAttribute('style', `height:${upContainer.height}px`);
+    }
+
+    this.calculate();
+  }
+
```

Comment:
If possible it would be good to use ember to get to these DOM elements instead of using `querySelector`. I'm thinking if you move the modifiers to the `#-container` elements themselves, that will pass the elements through as arguments meaning you can avoid using `querySelector` in order to find those.

Then I think if you use a `{{did-insert (set this 'upstream/downstreamLines')}}` on the lines DOM elements that should give you a reference on the component for each of the lines DOM elements, meaning you can avoid `querySelector` for those also.

Not a necessity but would make this a little nicer by depending less on ids. I think if thats ^ not possible for whatever reason we should at least cache the results of `querySelector` so we aren't doing it multiple times.